packages/network/src/connection/WebRtcEndpoint.ts

```diff
@@ -378,11 +378,16 @@ export class WebRtcEndpoint extends EventEmitter implements IWebRtcEndpoint {
             this.rtcSignaller.sendRtcConnect(routerId, connection.getPeerId())
         }
 
-        const deferredAttempt = connection.getDeferredConnectionAttempt()
+        const deferredAttempt = connection.getDeferredConnectionAttempt() 
+        
+        if (connection.getLastState() == 'connected') {
+            return Promise.resolve(targetPeerId)
```

Comment:
Perhaps eslint/typescript is complaining but you shouldn't need to wrap returns in `Promise.resolve`. 
`return targetPeerId` will do the same thing as `return Promise.resolve(targetPeerId)` inside an `async` function.